### ISSUE TITLE
Avoid prototype injection

### ISSUE THREAD

**Issue by Tyriar** [microsoft/vscode-jupyter#12020]
I noticed these:

https://github.com/microsoft/vscode-jupyter/blob/890dd7a885cff5210273be38698162021eeabe90/src/platform/common/extensions.ts

https://github.com/microsoft/vscode-jupyter/blob/890dd7a885cff5210273be38698162021eeabe90/src/webviews/webview-side/error-renderer/index.ts#L55-L60

This is not good practice as it changes the global state and it also unexpected. Reading this code for the first time I googling `String.format` assuming you were polyfilling it. These could just be regular functions that you could import like `import { format } from '.../string-utils';`

**Comment by DonJayamanne** [comment]
Agreed, the plan was to replace this, however most felt it was unnecessary.
We have found that using this caused other issues, Python extension had their own implementation as well and that caused issues with our extension, however since they ended up using imports instead of prototype, the change to our code was considered unnecessary.

TLDR: I'm all for making this change.
Including removing other prototypes like `ignoreErrors` which can be simply fixed with `.catch(noop)`

**Comment by DonJayamanne** [comment]
Here's the original issue https://github.com/microsoft/vscode-jupyter/issues/9607

**Comment by Tyriar** [comment]
/duplicate https://github.com/microsoft/vscode-jupyter/issues/9607
